homeassistant/components/onvif/parsers.py

```diff
@@ -0,0 +1,367 @@
+# pylint: disable=protected-access
+async def async_parse_storage_failure(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:Device/HardwareFailure/StorageFailure
+    Async Friendly.
+    """
+    try:
+        source = msg.Message._value_1.Source.SimpleItem[0].Value
+        return Event(
+            f"{uid}_{msg.Topic._value_1}_{source}",
+            "Storage Failure",
+            "binary_sensor",
+            "problem",
+            None,
+            msg.Message._value_1.Data.SimpleItem[0].Value == "true",
+        )
+    except (AttributeError, KeyError):
+        return None
+
+
+@PARSERS.register("tns1:Monitoring/ProcessorUsage")
+# pylint: disable=protected-access
+async def async_parse_processor_usage(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:Monitoring/ProcessorUsage
+    Async Friendly.
+    """
+    try:
+        usage = float(msg.Message._value_1.Data.SimpleItem[0].Value)
+        if usage <= 1:
+            usage *= 100
+
+        return Event(
+            f"{uid}_{msg.Topic._value_1}",
+            "Processor Usage",
+            "sensor",
+            None,
+            "percent",
+            int(usage),
+        )
+    except (AttributeError, KeyError):
+        return None
+
+
+@PARSERS.register("tns1:Monitoring/OperatingTime/LastReboot")
+# pylint: disable=protected-access
+async def async_parse_last_reboot(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:Monitoring/OperatingTime/LastReboot
+    Async Friendly.
+    """
+    try:
+        return Event(
+            f"{uid}_{msg.Topic._value_1}",
+            "Last Reboot",
+            "sensor",
+            "timestamp",
+            None,
+            dt_util.as_local(
+                dt_util.parse_datetime(msg.Message._value_1.Data.SimpleItem[0].Value)
+            ),
+        )
+    except (AttributeError, KeyError):
+        return None
+
+
+@PARSERS.register("tns1:Monitoring/OperatingTime/LastReset")
+# pylint: disable=protected-access
+async def async_parse_last_reset(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:Monitoring/OperatingTime/LastReset
+    Async Friendly.
+    """
+    try:
+        return Event(
+            f"{uid}_{msg.Topic._value_1}",
+            "Last Reset",
+            "sensor",
+            "timestamp",
+            None,
+            dt_util.as_local(
+                dt_util.parse_datetime(msg.Message._value_1.Data.SimpleItem[0].Value)
+            ),
+        )
+    except (AttributeError, KeyError):
+        return None
+
+
+@PARSERS.register("tns1:Monitoring/OperatingTime/LastClockSynchronization")
+# pylint: disable=protected-access
+async def async_parse_last_clock_sync(uid: str, msg) -> Event:
+    """Handle parsing event message.
+
+    Topic: tns1:Monitoring/OperatingTime/LastClockSynchronization
+    Async Friendly.
+    """
+    try:
+        return Event(
```

Comment:
This event should result in an entity that is disabled by default. Should that be part of the event?

Reply:
We can put it there since I’m basically passing that around. To clarify, you are talking specifically about the clock sync entity disabled by default and not all of them, correct?

Reply:
Yeah. Probably last reset is also not necessary?